fabric-crafting-v1/src/main/java/net/fabricmc/fabric/api/crafting/v1/FabricRecipeRemainder.java

```diff
@@ -0,0 +1,18 @@
+package net.fabricmc.fabric.api.crafting.v1;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.inventory.CraftingInventory;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+
+public interface FabricRecipeRemainder {
```

Comment:
You should add a doc here saying what classes (`Item` subclasses in this case) should implement this.

Reply:
Imo we should just add per-recipe remainders than this general form of remainder (though remainers might be controlled by custom recipe types?)